lib/graphhopper/api_client.rb

```diff
@@ -277,6 +278,11 @@ def build_request_body(header_params, form_params, body)
       else
         data = nil
       end
+
+      if data && header_params['Content-Encoding'] == 'gzip'
```

Comment:
I think that setting the header in one place and then reading it another to make a decision is a bit confusing and makes the code hard to follow.

What if we had a `use_compression` option for `GraphHopper:: ApiClient` that defaults to `false`? Then `GraphHopper:: VrpApi` can set it to true in `initialize`. I find the pattern of calling `select_header_content_type` a bit confusing, but we could follow it and implement `select_header_content_encoding` that returns the correct value based on `use_compression`. If we do this, all the compression logic is encapsulated in `GraphHopper:: ApiClient`.

What do you think?